lib/efi.c

```diff
@@ -284,10 +292,25 @@ static uint64_t efi_set_mem_regions(UINTN *MapKey)
 		case EfiACPIMemoryNVS:
 		case EfiPalCode:
 		case EfiBootServicesCode:
-		case EfiBootServicesData:
 		case EfiRuntimeServicesCode:
 			r->flags = MR_F_RESERVED;
 			break;
+		case EfiBootServicesData:
```

Comment:
I don't believe BootServicesData is a good memory region type to use nor that this is the best way to allocate the memory from there (we could just use AllocatePool after all). We probably want to use uefi_call_wrapper(BS->AllocatePool, 3, EfiRuntimeServicesData, ...) instead.

Reply:
That's a very good point and I've changed it. I read in the manual that after exiting boot services we're allowed to reuse the allocated space but what you're suggesting sounds much better.